### PR TITLE
modified newCarsCreator controller

### PR DESCRIPTION
**Pull request by tarasulo**

### CODE REVIEW

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package tkhal.filter.controller;
+
+import model.Car;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CarsFilter {
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: Add java docs for each method and class

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(MessageFilterController.class);
+
+    public Car filter(Car tempCar) {
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: I would suggest to return `boolean` instead for `Car`

File: carsFilter/src/main/java/tkhal.filter.controller/FilterKafkaProducer.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private String topicFilteredName;
+    private final static Logger LOGGER = LoggerFactory.getLogger(FilterKafkaProducer.class);
+    private Producer <String, Car> producer;
+
+    public Producer <String, Car> createProducer() {
+        Properties propsRedirect = new Properties();
+
+        //reading Kafka producer properties from config file
+        try {
+            ClassLoader classloader = Thread.currentThread().getContextClassLoader();
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/FilterKafkaProducer.java]: Duplicated piece of code: see `FilterKafkaConsumer` lines 22-25 and this class lines 24-27. Refactor and extract common piece

File: carsReceiver/src/main/java/tkhal.receiver.controller/AfterFilterConsumer.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    private String topicName;
+    private final static Logger LOGGER = LoggerFactory.getLogger(CarStandardizerController.class);
+
+    public KafkaConsumer<String, Car> startConsumer() {
+
+        // reading Kafka consumer properties from config file
+        Properties props = new Properties();
+        try {
+            ClassLoader classloader = Thread.currentThread().getContextClassLoader();
```
Comment (kolyansv) [inline carsReceiver/src/main/java/tkhal.receiver.controller/AfterFilterConsumer.java]: Duplicated code - see comments above

File: carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+public class MessageFilterController {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(MessageFilterController.class);
+    private static Car tempCar = null;
+    private static KafkaConsumer<String, Car> consumer;
+    private static Producer<String, Car> producer;
+    private static FilterKafkaConsumer filterKafkaConsumer = new FilterKafkaConsumer();
+    private static FilterKafkaProducer filterKafkaProducer = new FilterKafkaProducer();
+    private static CarsFilter carsFilter = new CarsFilter();
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java]: very bad practice to instantiate classes like this.
Use constructor for it

File: carsReceiver/src/main/java/tkhal.receiver.controller/CarStandardizerController.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import java.time.Duration;
+
+public class CarStandardizerController {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(CarStandardizerController.class);
+    private static Car standardizerCar = null;
+    private static AfterFilterConsumer afterFilterConsumer = new AfterFilterConsumer();
+    private static KafkaConsumer<String, Car> consumer;
+    private static JdbcCarsWriter jdbcCarsWriter = new JdbcCarsWriter();
+    private static CarStandardizer standardizer = new CarStandardizer();
```
Comment (kolyansv) [inline carsReceiver/src/main/java/tkhal.receiver.controller/CarStandardizerController.java]: Do not instantiate classes like this. Use constructor

**Review by kolyansv — CHANGES_REQUESTED** [review]

File: carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java (hunk 5 above)
Comment (tarasulo) [reply carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java]: done

File: carsReceiver/src/main/java/tkhal.receiver.controller/CarStandardizerController.java (hunk 6 above)
Comment (tarasulo) [reply carsReceiver/src/main/java/tkhal.receiver.controller/CarStandardizerController.java]: done

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java (hunk 2 above)
Comment (tarasulo) [reply carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: done

File: carsFilter/src/main/java/tkhal.filter.controller/FilterKafkaProducer.java (hunk 3 above)
Comment (tarasulo) [reply carsFilter/src/main/java/tkhal.filter.controller/FilterKafkaProducer.java]: done

File: carsReceiver/src/main/java/tkhal.receiver.controller/AfterFilterConsumer.java (hunk 4 above)
Comment (tarasulo) [reply carsReceiver/src/main/java/tkhal.receiver.controller/AfterFilterConsumer.java]: done

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java (hunk 1 above)
Comment (tarasulo) [reply carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: done

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java [hunk 7]
```diff
@@ -4,17 +4,25 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * @author Taras Khalak
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: Will this information help other developer to understand what does this class do?

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java [hunk 8]
```diff
@@ -4,17 +4,25 @@
...
 public class CarsFilter {
-
+    /**
+     * This is the class for filtering cars
+     */
     private final static Logger LOGGER = LoggerFactory.getLogger(MessageFilterController.class);
 
-    public Car filter(Car tempCar) {
-        // filtering car
+    public Boolean filter(Car tempCar) {
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: why not to use primitive?
Keep in mind: prefer using primitives when it possible

File: carsFilter/src/main/java/tkhal.filter.controller/FilterKafkaConsumer.java [hunk 9]
```diff
@@ -4,28 +4,35 @@
...
 import java.util.Collections;
 import java.util.Properties;
-
+/**
+ * @author Taras Khalak
+ */
 public class FilterKafkaConsumer {
+    /**
+     * This is the class for creating Kafka Consumer
+     */
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/FilterKafkaConsumer.java]: Does this comment belong to `topicName` variable?

**Review by kolyansv — CHANGES_REQUESTED** [review]
Did you read any docs how to write java docs properly?

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java (hunk 8 above)
Comment (tarasulo) [reply carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: done

File: carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java (hunk 7 above)
Comment (tarasulo) [reply carsFilter/src/main/java/tkhal.filter.controller/CarsFilter.java]: done

File: carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java [hunk 10]
```diff
@@ -32,11 +29,12 @@ public MessageFilterController() {
...
+     * This is the main method which getting cars, filtration them,
+     * and writing to the new Kafka topic
+     */
     public static void main(String[] args) {
-        /**
-         * This is the main method which getting cars, filtration them,
-         * and writing to the new Kafka topic
-         */
+
         new MessageFilterController();
```
Comment (kolyansv) [inline carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java]: this is bad!
extract everything from `main` to another method
`main` should create instance of class and execute that method on that instance
apply for all mains

**Review by kolyansv — CHANGES_REQUESTED** [review]

File: carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java (hunk 10 above)
Comment (tarasulo) [reply carsFilter/src/main/java/tkhal.filter.controller/MessageFilterController.java]: done

**Review by kolyansv — APPROVED** [review]

**Pull request merged**